### PR TITLE
debian stretch slim

### PR DESCRIPTION
**Pull request by mwlczk**
switch to debian:stretch-slim:
 - first step correct the testdata, as newer packages are more strict
about the mail-structure.

 - add missing build-step in `make`
 - clean the userdb aswell
 - use timeout of netcat (nc -w), as postgrey would not close the connection
 - there is 2 extra mail-log-files (mail.info and mail.warn) -> count only mail.log
 - cosmetic: use "" instead of ''

new image benefits:
 - smaller size (52MB less)
 - 0 CVEs compared to 11 CVEs in ubuntu 16.04 Image

better backport situation:
 - postfix 3.1.6 vs 3.1.0
 - fail2ban 0.9.6 vs 0.9.3
 - openssl 1.1.0f vs 1.0.2g
 - dovecot 2.2.27 vs 2.2.22
 ....

changes in Dockerfile and configfiles needed because of stretch-slim:
- add missing gnupg and iproute2 package
- remove non-free rar, unrar-free should do
- rsyslog does not add syslog user and has different conf-structure
- pyzor command discover was deprecated and is missing in the new
stretch package

- dovecot does not know SSLv2 anymore. removed because of warnings in
log

- iptables does not know imap3, IMAP working group chose imap2 in favor
of imap3

- s. following commits aswell

### CODE REVIEW

**Comment by mwlczk** [comment]
The tests run fine on my machine. Need to look into it, why Travis breaks. They changed the architecture mid-December though. Any hints on that would be great

**Comment by mwlczk** [comment]
retrigger travis ci to get a closer look into the build.

**Comment by mwlczk** [comment]
this is odd: travis might have some caching (or no-cache) issues.
First build run had 9 failures.
Second build run had only 1 failure - but no source code has been changed in between.

Gonna hunt that last failure down now.

**Comment by mwlczk** [comment]
One more Travis try

**Comment by mwlczk** [comment]
Last two Travis runs exit at different tests.

**Comment by mwlczk** [comment]
Travis makes me wonder. If I seemingly fix one bug/test, it fails another test, which is executed earlier in the queue and passed all the times before.

**Comment by johansmitsnl** [comment]
@mwlczk First of great PR!

A very good contribution, myself also tried to change this image to Debian because of better backport support that lacks in Ubuntu sometimes, couldn't agree more.

Sometimes travis is busy and due to the amount of containers that are started some tests are not yet ready to be tested and this gives sometimes some mixed test results. This is a improvement that needs to be applied to this project #324.

Looking forward to merging this PR but some questions:

* Could you remove commented old code?
* Besides the tests are you using the PR live?
* Any notes that should be considered when merging, todo's or ...?

File: Dockerfile [hunk 1]
```diff
@@ -60,7 +62,7 @@ RUN apt-get update -q --fix-missing && \
     postfix-pcre \
     postfix-policyd-spf-python \
     pyzor \
-    rar \
+    # rar \
```
Comment (johansmitsnl) [inline Dockerfile]: Remove the line if not used

File: Dockerfile [hunk 2]
```diff
@@ -177,8 +180,9 @@ RUN sed -i -r "/^#?compress/c\compress\ncopytruncate" /etc/logrotate.conf && \
   chown -R clamav:root /var/log/mail/clamav.log && \
   touch /var/log/mail/freshclam.log && \
   chown -R clamav:root /var/log/mail/freshclam.log && \
-  sed -i -r 's|/var/log/mail|/var/log/mail/mail|g' /etc/rsyslog.d/50-default.conf && \
-  sed -i -r 's|;auth,authpriv.none|;mail.none;mail.error;auth,authpriv.none|g' /etc/rsyslog.d/50-default.conf && \
+  # no separate 50-default.conf
```
Comment (johansmitsnl) [inline Dockerfile]: Remove the line if not used

File: test/tests.bats [hunk 3]
```diff
@@ -166,8 +166,8 @@ load 'test_helper/bats-assert/load'
 }
 
 @test "checking postgrey: there should be a log entry about the whitelisted and passed e-mail user@whitelist.tld in /var/log/mail/mail.log" {
-  run docker exec mail_with_postgrey /bin/sh -c "nc 0.0.0.0 10023 < /tmp/docker-mailserver-test/nc_templates/postgrey_whitelist.txt"
-  sleep 8
+  run docker exec mail_with_postgrey /bin/sh -c "nc -w 8 0.0.0.0 10023 < /tmp/docker-mailserver-test/nc_templates/postgrey_whitelist.txt"
+  # sleep 8
```
Comment (johansmitsnl) [inline test/tests.bats]: Remove the line if not used

**Comment by mwlczk** [comment]
@johansmitsnl
- removed commented old code
- am using this PR live
- Notes: We should have a look at the cipher-suites for all 
  TLS-Endpoints (smtp, smtpd, imaps, pop3s, ...), because newer
  package (e.g. openssl) might cause problems with older stuff. 
  Anyway we should keep an eye to update them in a next PR.

**Pull request merged**